### PR TITLE
docs: update examples for onboarding_cli

### PR DESCRIPTION
**Pull request by srieteja**
**- What I did**
- ensure onboarding_cli is terminated completely after closing
- adopt arg_parser in examples
- move legacy examples to onboarding_cli/examples/legacy_examples
- move apkam examples to onboarding_cli/examples/apkam_examples
- Introduce a custom_arg_parser that validates arguments passed to the arg_parser
- Updated the examples readme.md

**- How I did it**
- [IMP] added an exit(0) to OnboardingService.close() to ensure onboarding_service is terminated completely and successfully

**- Description for the changelog**

fix: add exit(0) to OnboardingService.close()

### CODE REVIEW

File: packages/at_onboarding_cli/example/README.md [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
-   e.g. dart example/onboard.dart @alice🛠 /home/alice/.atsign/@alice🛠_wavikey.atKeys
-2. Authenticate using the onboarded atsign
-   dart example/apkam_authenticate.dart <atsign> <path_of_keys_file_from_#1>
-   e.g. dart example/apkam_authenticate.dart @alice🛠 /home/alice/.atsign/@alice🛠_wavikey.atKeys
-3. Run client to approve enrollments
-   dart example/enroll_app_listen.dart <atsign> <path_of_keys_file_from_#1>
-   e.g dart example/enroll_app_listen.dart @alice🛠 /home/alice/.atsign/@alice🛠_wavikey.atKeys
+1. Onboard an atsign which has privilege to approve/deny enrollments:<br>
+   - run: `dart example/onboard.dart -a <atsign> -c <CRAM secret> -k <path_store_keys_file>`<br>
+   - e.g. `dart example/onboard.dart -a @alice -k /home/alice/.atsign/@alice_wavikey.atKeys`<br/>
```
Comment (murali-shris) [inline packages/at_onboarding_cli/example/README.md]: e.g should include -c <cram_secret>

**Review by murali-shris — CHANGES_REQUESTED** [review]

**Comment by murali-shris** [comment]
* Fix the analyzer issue:
warning - example/apkam_examples/apkam_enroll.dart:1:8 - Unused import: 'dart:io'. Try removing the import directive. - unused_import
* check pub score 
* modify pubspec and changelog for publishing

File: packages/at_onboarding_cli/example/README.md (hunk 1 above)
Comment (srieteja) [reply packages/at_onboarding_cli/example/README.md]: change made as part of [fc488f0](https://github.com/atsign-foundation/at_libraries/pull/442/commits/fc488f0742ad967a406ba94aea50a2dc7cfbb1b6)

**Comment by srieteja** [comment]
> * Fix the analyzer issue:
>   warning - example/apkam_examples/apkam_enroll.dart:1:8 - Unused import: 'dart:io'. Try removing the import directive. - unused_import
> * check pub score
> * modify pubspec and changelog for publishing

- Analyzer issues have been fixed
- Pub score is now 130/130
- Changelog has been modified to reflect all relevant changes

**Review by murali-shris — APPROVED** [review]

**Pull request merged**